src/zenml/integrations/kubeflow/flavors/kubeflow_orchestrator_flavor.py

```diff
@@ -84,13 +87,11 @@ def _validate_and_migrate_pod_settings(
             AssertionError: If old and new settings are used together.
             ValueError: If username and password are not specified together.
         """
-        has_pod_settings = bool(values.get("pod_settings"))
+        has_pod_settings = bool(data.get("pod_settings"))
 
-        node_selectors = cast(
-            Dict[str, str], values.get("node_selectors") or {}
-        )
+        node_selectors = cast(Dict[str, str], data.get("node_selectors") or {})
         node_affinity = cast(
-            Dict[str, List[str]], values.get("node_affinity") or {}
+            Dict[str, List[str]], data.get("node_affinity") or {}
```

Comment:
`node_affinity` is already `{}` by default

Reply:
I think with the upcoming [Kubeflow, Tekton, and Vertex PR](https://github.com/zenml-io/zenml/pull/2733/files#diff-7c9ffb50ca2198c7533a60b1c699ecca0944711f6f434f11cf088c699a8adbdc), this will anyhow change.